tests/tests.rs

```diff
@@ -62,6 +62,24 @@ fn test_input_as_slice_less_safe() {
     assert_eq!(input.as_slice_less_safe(), slice);
 }
 
+#[test]
+fn test_input_equality() {
+    let slice1 = b"foo";
```

Comment:
In general I prefer to have tests that are "parameterized"; that is, where there is an explicit list of test inputs (and maybe expected result for each one, if the expected result isn't the same for all of them), where the test loops over the list of inputs. This way it is clearer to see what is being tested for each input. See recent PRs for an example of this style.